src/HexagonOptimize.cpp

```diff
@@ -1169,20 +1123,6 @@ class OptimizePatterns : public IRMutator {
         return visit_let<Stmt>(op);
     }
 
-    Expr visit(const Div *op) override {
-        if (!op->type.is_float() && op->type.is_vector()) {
-            return mutate(lower_int_uint_div(op->a, op->b));
-        }
-        return IRMutator::visit(op);
-    }
-
-    Expr visit(const Mod *op) override {
-        if (!op->type.is_float() && op->type.is_vector()) {
-            return mutate(lower_int_uint_mod(op->a, op->b));
-        }
-        return IRMutator::visit(op);
-    }
-
```

Comment:
@dsharletg Do we not want to use long division here? I see degradations in mul_div_mod because of this?

Master

Testing div_mod vector_width: 128
Hexagon simulator executed function 0x91af0 in 546000 cycles
Hexagon simulator executed function 0x98530 in 2262000 cycles
Hexagon simulator executed function 0xaae50 in 15386000 cycles
Hexagon simulator executed function 0x94a60 in 978000 cycles
Hexagon simulator executed function 0xb1590 in 4468000 cycles
Hexagon simulator executed function 0xca5f0 in 24041000 cycles

Here:

Testing div_mod vector_width: 128
Hexagon simulator executed function 0xa1050 in 21911000 cycles
Hexagon simulator executed function 0xaacc0 in 23752000 cycles
Hexagon simulator executed function 0xb42c0 in 24363000 cycles
Hexagon simulator executed function 0xcebf0 in 50166000 cycles
Hexagon simulator executed function 0xef550 in 55319000 cycles
Hexagon simulator executed function 0xd8ee0 in 58044000 cycles

Reply:
At one point, I had moved this to FindIntrinsics, but reverted that. Fixed.